pkg/analyzer/lib/src/util/fast_uri.dart

```diff
@@ -86,6 +86,12 @@ class FastUri implements Uri {
   bool get isAbsolute => hasScheme;
 
   @override
+  bool isScheme(String scheme) {
+    if (scheme == null || scheme.isEmpty) return !hasScheme;
```

Comment:
style nit: within the analyzer code base we always have blocks around the `then` and `else` statements.